assignments/array-methods.js

```diff
@@ -56,22 +56,40 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+runners.forEach(function(element) {
+    let runnerFullName = (element.first_name + element.last_name);
+    fullName.push(runnerFullName);
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+runners.forEach(function(element) {
+    let capitalizedName = (element.first_name.toUpperCase());
+    allCaps.push(capitalizedName);
+});
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+const largeShirtsFilter = runners.filter((data) =>{
+    if (data.shirt_size == 'L') {
+        largeShirts.push (data);
+    }
+});
+return largeShirtsFilter;
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
-console.log(ticketPriceTotal);
+
+const ticketPriceTotal = runners.reduce((theReducer, item) =>{
+    return theReducer += item.donation;
```

Comment:
No need for `+=`, `+` works here because the reducer method is already aggregating the results of the return.